Add types for oidc.id_token.pre_create

louischan-oursky opened the pull request:
ref DEV-3060

louischan-oursky committed:
Add types for oidc.id_token.pre_create

tung2744 commented on mod.ts:
```diff
@@ -390,6 +390,19 @@ export type EventOIDCJWTPreCreateHookResponse =
+  payload: {
+    user: User;
+    identities: Identity[];
+    jwt: JWT;
```
@louischan-oursky Should we rename this to `id_token`? As discussed before the name `jwt` is confusing.

louischan-oursky replied:
Addressed.

tung2744 commented on tests/main.test.ts:
```diff
@@ -240,3 +244,33 @@ async function oidc_jwt_pre_create(
+  return {
+    is_allowed: true,
+    mutations: {
+      jwt: {
```
Should this also be id_token?

louischan-oursky replied:
Addressed.

louischan-oursky committed:
Correct some typos

louischan-oursky committed:
Rename jwt to id_token

Also improve the type definition so that blocking hook response
now can only contain supported mutations.
